testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/ServerEndpoint.java

```diff
@@ -8,39 +8,54 @@
 import io.opentelemetry.api.trace.Span;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.Arrays;
+import java.util.HashMap;
 import java.util.Map;
-import java.util.stream.Collectors;
-
-public enum ServerEndpoint {
-  SUCCESS("success", 200, "success"),
-  REDIRECT("redirect", 302, "/redirected"),
-  ERROR("error-status", 500, "controller error"), // "error" is a special path for some frameworks
-  EXCEPTION("exception", 500, "controller exception"),
-  NOT_FOUND("notFound", 404, "not found"),
-  CAPTURE_HEADERS("captureHeaders", 200, "headers captured"),
-  NESTED_PATH(
-      "nestedPath/hello/world",
-      200,
-      "nested path"), // TODO (heya) move it to webflux test module after this has been converted to
-  // a class
-  CAPTURE_PARAMETERS("captureParameters", 200, "parameters captured"),
+
+public class ServerEndpoint {
+  private static final Map<String, ServerEndpoint> PATH_MAP = new HashMap<>();
+  public static final ServerEndpoint SUCCESS =
+      new ServerEndpoint("SUCCESS", "success", 200, "success");
+  public static final ServerEndpoint REDIRECT =
+      new ServerEndpoint("REDIRECT", "redirect", 302, "/redirected");
+  public static final ServerEndpoint ERROR =
+      new ServerEndpoint(
+          "ERROR",
+          "error-status",
+          500,
+          "controller error"); // "error" is a special path for some frameworks
+  public static final ServerEndpoint EXCEPTION =
+      new ServerEndpoint("EXCEPTION", "exception", 500, "controller exception");
+  public static final ServerEndpoint NOT_FOUND =
+      new ServerEndpoint("NOT_FOUND", "notFound", 404, "not found");
+  public static final ServerEndpoint CAPTURE_HEADERS =
+      new ServerEndpoint("CAPTURE_HEADERS", "captureHeaders", 200, "headers captured");
+  public static final ServerEndpoint NESTED_PATH =
+      new ServerEndpoint("NESTED_PATH", "nestedPath/hello/world", 200, "nested path");
+  // TODO (heya) move it to webflux test module after this has been converted to a class
```

Comment:
I think this comment applies to `NESTED_PATH` not `CAPTURE_PARAMETERS`

Reply:
```suggestion
  // TODO (heya) move it to webflux test module after this has been converted to a class
  public static final ServerEndpoint NESTED_PATH =
      new ServerEndpoint("NESTED_PATH", "nestedPath/hello/world", 200, "nested path");
```